### PR TITLE
Added aniticipation effect on sheet with a bindable property

### PR DESCRIPTION
**Pull request by haavamoa**
## Added / Fixed

- Added aniticipation effect on sheet with a bindable property

## Todo List

- [X] I have tested on an Android device.
- [x] I have tested on an iOS device.
- [ ] I have added unit tests

>*Please add pictures / Gifs if possible*
>*Todo List can be checked by putting a `X` inside the brackets*
>*Remember to update our `wiki` after this PR is merged*

### CODE REVIEW

**Comment by haavamoa** [comment]
Not tested on iOS device yet.

**Comment by haavamoa** [comment]
Missing checks of if it should use the effect when the sheet is too big. Looks silly if it is full screened.

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs [hunk 1]
```diff
@@ -441,8 +456,19 @@ public async Task BeforeRemoval()
...
+            var length = 250;
+            Task translationTask = Task.CompletedTask;
+            if (HasAnticipationEffect)
+            {
+                length = 600;
+                translationTask = m_sheetView.SheetFrame.TranslateTo(m_sheetView.SheetFrame.X, y, (uint)length, Easing.SpringIn);
+            }
+            else
+            {
+                translationTask = m_sheetView.SheetFrame.TranslateTo(m_sheetView.SheetFrame.X, y);
```
Comment (Illedan) [inline src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: This length is 250 right? So why is the Task.Delay 600 and not length?

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs [hunk 2]
```diff
@@ -555,7 +581,7 @@ private async void ToggleSheetVisibility()
                 OnBeforeCloseCommand?.Execute(OnBeforeCloseCommandParameter);
                 OnBeforeClose?.Invoke(this, EventArgs.Empty);
 
-                m_modalityLayout.Hide(m_sheetView.SheetFrame);
+                m_modalityLayout.Hide(m_sheetView.SheetFrame, 600);
```
Comment (Illedan) [inline src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: 600 for all cases? Or 250 without Anticipation?

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs (hunk 1 above)
Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: Length and task await should be 600 when anticipation is enabled

Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: I'll clean up when I will continue it the branch.

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs (hunk 2 above)
Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: 250 without.

Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: Fix'd

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs [hunk 3]
```diff
@@ -441,8 +456,19 @@ public async Task BeforeRemoval()
...
+            {
+                length = 600;
+                translationTask = m_sheetView.SheetFrame.TranslateTo(m_sheetView.SheetFrame.X, y, (uint)length, Easing.SpringIn);
+            }
+            else
+            {
+                translationTask = m_sheetView.SheetFrame.TranslateTo(m_sheetView.SheetFrame.X, y);
+            }
+            
+            await Task.Delay(600);
```
Comment (Illedan) [inline src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: Still 600 when the if goes into the else?

**Comment by haavamoa** [comment]
After an internal demonstration and discussion we have ended up with providing the effect for when the sheet opens but not when it closes. this should be changed in this PR.

**Comment by haavamoa** [comment]
#ios
anticipation_ios

**Comment by haavamoa** [comment]
This PR should not be merged. Me and @darkwood has discussed the anticipation effect.

It should only "bounce" when the sheet has snapping + acceleration when people drag it. You do not anticipate it to animate like it does in this PR when the sheet opens programmatically.

**Pull request closed without merging**